src/scipp/core/shape.py

```diff
@@ -101,23 +124,39 @@ def concat(x: Sequence[VariableLike], dim: str) -> VariableLike:
     return _call_cpp_func(_cpp.concat, x, dim)
 
 
-def fold(x: VariableLike,
+def fold(x: VariableLikeType,
          dim: str,
          sizes: Optional[Dict[str, int]] = None,
          dims: Optional[Union[List[str], Tuple[str, ...]]] = None,
-         shape: Optional[Sequence[int]] = None) -> VariableLike:
+         shape: Optional[Sequence[int]] = None) -> VariableLikeType:
     """Fold a single dimension of a variable or data array into multiple dims.
 
-    :param x: Variable or DataArray to fold.
-    :param dim: A single dim label that will be folded into more dims.
-    :param sizes: A dict mapping new dims to new shapes.
-    :param dims: A list of new dims labels.
-    :param shape: A list of new dim shapes.
-    :raises: If the volume of the old shape is not equal to the
-             volume of the new shape.
-    :return: Variable or DataArray with requested dimension labels and shape.
-
-    Examples:
+    Parameters
```

Comment:
I think this also needs the comment "One and only one of these sets of arguments must be given:"